reference/compiling_for_android.rst

```diff
@@ -16,93 +16,81 @@ Also, you still need to do all the steps mentioned in the
 :ref:`doc_exporting_for_android` tutorial before attempting your custom
 export template.
 
-Requirements
-------------
+A. Requirements
+---------------
 
 For compiling under Windows, Linux or OSX, the following is required:
 
--  Python 2.7+ (3.0 is untested as of now).
--  SCons build system.
+-  Python 2.7+ [3.0 is untested as of now]
+-  pywin32 [Optional, Windows only. Applied when -j(number of cores) specified]
+-  SCons build system
 -  Android SDK version 19 [Note: Please install all Tools and Extras of sdk manager]
 -  Android build tools version 19.1
--  Android NDK
--  Gradle
--  OpenJDK 6 or later (or Oracle JDK 6 or later)
+-  Android NDK r10e or later
+-  Gradle [will be downloaded and installed when compiling for android if it doesn't exist]
+-  JDK 6+ (Oracle's releases) or OpenJDK 6+
+
+B. Setting up Build System
+--------------------------------------------------------
+
+1) Scons : please, refer to their respective docs
 
-Setting up SCons
-----------------
+2) Gradle : will be automatically setup upon packing APK if 
+we do not have gradle installed previously (may require working internet connection)
+
+3) Android :
 
 Set the environment variable ANDROID_HOME to point to the Android
 SDK.
 
 Set the environment variable ANDROID_NDK_ROOT to point to the
 Android NDK.
 
-To set those environment variables on Windows, press Windows+R, type
-"control system", then click on **Advanced system settings** in the left
-pane, then click on **Environment variables** on the window that
-appears.
+- on Windows :
+press Windows+R, type "control system", then click on **Advanced system settings**
+in the left pane, then click on **Environment variables** on the window that appears.
 
-To set those environment variables on Linux, use
+- on Unix (eg. Linux and Mac OS):
+Open up Terminal and then execute commands below :
 ``export ANDROID_HOME=/path/to/android-sdk`` and
-``export ANDROID_NDK_ROOT=/path/to/android-ndk``.
+``export ANDROID_NDK_ROOT=/path/to/android-ndk``
+
+C. Compiling
+------------
 
-Compiling
----------
+C1. Compiling the Engine
+------------------------
 
 Go to the root dir of the engine source code and type:
 
 ::
 
     C:\godot> scons platform=android
 
-This should result in a regular .so in ``\bin`` folder as if it was
+This should result in a regular .so file (built for ARM platform) in ``\bin`` folder, as if it was
 compiled with flags: ``tools=no target=debug``. The resulting file will
-be huge because it will contain all debug symbols, so for next builds,
-using ``target=release_debug`` or ``target=release`` is recommended.
-
-Copy the .so to the ``libs/armeabi`` Android folder (or symlink if you are
-in Linux or OSX). Note: Git does not support empty directories so you
-will have to create it if it does not exist:
+be huge because it will contain all debug symbols.
```

Comment:
Well as of now the engine (`tools=yes`) cannot be built for Android, only the export templates (`tools=no`).

Removing the information about copying the .so is good of course, but what I complain about is removing ", so for next builds, using ``target=release_debug`` or ``target=release`` is recommended." and keeping only "The resulting file will be huge because it will contain all debug symbols."

As it stands now, we just tell users:
- build with default parameters
- yeah it's use because it contains debug info, sorry about that! :)

So we should either tell them:

Version 1 (what was before)

- build with default parameters
- yeah it's use because it contains debug info, sorry about that! :)
- now use these custom parameters to actually build what you want (release and release_debug templates)

Version 2 (what would be better)

- build with those custom parameters to get the release and debug templates